legacy/mystery_game_nightmare27.py

```diff
@@ -0,0 +1,130 @@
+            self.guessed_letters.append(player_guess)
+            guess_result = self.calculate_new_word_list(player_guess)
+            if guess_result:
+                return 1
+            else:
+                self.remaining_guesses -= 1
+                return 0
+
+    def check_win_lose(self):
+        """ Checks to see if the player has won or lost the game """
+
+        if not self.remaining_guesses:
+            self.word = random.choice(self.possible_words)
+            return 1
+        elif "_" not in self.evasive_word:
+            return 2
+        return 0
+
+    def choose_word_class(self, class_list):
+        """Determines which word class gives the computer the highest chance
+        to win"""
+        best_class = "default"
+
+        for word_class in class_list:
+            if len(class_list[word_class]) > len(class_list[best_class]):
+                best_class = word_class
+
+        self.possible_words = class_list[best_class]
+
+        return best_class
+
+    def determine_class_types(self, player_guess):
+        """Examines words in list, returns class types based on location of
+        occurences of the player's guess in the string"""
+        class_type_list = {"default": []}
+
+        for word in self.possible_words:
+            class_title = ""
+            for letter_count in range(self.word_length):
+                if word[letter_count] == player_guess:
+                    class_title += "[" + str(letter_count) + "]"
+
+            if not len(class_title):
+                class_title = "None"
+
+            if class_title in class_type_list:
+                class_type_list[class_title].append(word)
+            else:
+                class_type_list[class_title] = [word]
+
+        return class_type_list
+
+    def load_dictionary(self, file):
+        """Loads the dictionary file from the specified location, file should
+        be in the format one word per line in the file"""
+        try:
+            with open(file, "r") as dictionary_file:
+                word_list = dictionary_file.read().split()
+            return word_list
+        except IOError:
+            file = self.default_dictionary
+            return self.load_dictionary(file)
+
+    def update_evasive_word(self, class_title, guess):
+        """Updates the evasive word to display any letters that are in the
+        word class that the computer player has chosen."""
+        parse_string = re.compile("(?<=\[)(\d+)(?=])")
+
+        indicies = re.findall(parse_string, class_title)
+
+        for index in indicies:
+            self.evasive_word[int(index)] = guess
+
+if __name__ == '__main__':
+    game = MysteryGameNightmare(file="dictionary.txt")
+    while True:
+        print(game.evasive_word)
+        print("{} Words left in pool".format(len(game.possible_words)))
+        guess = raw_input("Guess! : ")
+        if guess == "QUIT":
+            break
+        else:
+            game.check_guess(guess.upper())
```

Comment:
Not sure what's happening here. check_guess returns a number, but you aren't using it.